services/execsrv/store.go

```diff
@@ -199,10 +200,13 @@ func (p postgresStore) Update(id string, exec types.Exec) error {
 	panic("implement me")
 }
 
-func (p postgresStore) UpdateStatus(id string, status types.Status) error {
+// UpdateStatus updates exec status and relevant timestamps
+func (p postgresStore) UpdateStatus(id string, status types.Status, setReadyAt, setExitedAt time.Time) error {
 	params := db.ExecStatusUpdateParams{
-		ID:     id,
-		Status: db.UnweaveExecStatus(status),
+		ID:       id,
+		Status:   db.UnweaveExecStatus(status),
+		ReadyAt:  db.NullTimeFrom(setReadyAt),
+		ExitedAt: db.NullTimeFrom(setExitedAt),
```

Comment:
The underlying query uses `coalesce`. Does that work with the nullstring type? It doesn't overwrite the time if `setReadyAt` or `setExitedAt` is the zero value of `time.Time`?

Reply:
it only sets the first non-null value, timestamps won't get overwritten , see the e2e test:

![image](https://github.com/unweave/unweave/assets/8885269/af6c239e-1050-4b9e-9a5e-7e3189e2901f)